client/dashboard/app/routes/domain-routes.ts

```diff
@@ -44,8 +45,19 @@ export const siteDomainsRoute = createRoute( {
 export const domainRoute = createRoute( {
 	getParentRoute: () => rootRoute,
 	path: 'domains/$domainName',
-	// TODO: add the submenu nav components here
-} );
+	loader: async ( { params: { domainName } } ) => {
+		const domain = await queryClient.ensureQueryData( domainQuery( domainName ) );
+		if ( ! domain ) {
+			throw notFound();
+		}
```

Comment:
I think this case will never happen. The query will throw before it reaches this line.